lms/djangoapps/verify_student/management/commands/tests/test_send_verification_expiry_email.py

```diff
@@ -206,11 +206,14 @@ def test_not_enrolled_in_verified_course(self):
         self.assertEqual(len(mail.outbox), 1)
         self.assertIsNone(attempt.expiry_email_date)
 
+    @py2_only
     def test_user_enrolled_in_verified_course(self):
         """
         Test that if the user is enrolled in verified track, then after sending the default no of
         emails, `expiry_email_date` is updated to now() so that it's filtered in the future to send
-        email again
+        email again.
+
+        Does not work on python 3 with the latest mongo driver version due to class inheritance issues.
```

Comment:
Can you provide more details on this?  What is the actual error that we see?  Given that this is essentially deleting the test going forward, I want to understand what the issues are.

The `@py2_only` decorator was meant to be used during transition and is not meant to be permanent.

Reply:
@feanil Sorry, for the delay in replying, I missed this ping.
Using the new `pymongo` driver, the `ModuleStoreTestCase` doesn't correctly set up the module store for testing.
On the other set of tests, this was fixed by switching the inheritance order (see https://github.com/edx/edx-platform/pull/21197/files/5983d5dd42ca8f0f6e1347d04ccbcc4a28482a92#diff-16556c13da8f491bf5a44a4a628ccc25R26), but the workaroung didn't work on this one.
When I tried swapping the inheritance order here, the S3 portion of the code started failing, and this led me to believe there is something wrong with the class inheritance on the test cases classes (`ModuleStoreTestCase`, `MockS3Mixin`). 
I've spent some time on the issue and couldn't make the test work. :(

That said, the tests pass on Python 2, and the feature seems to be working fine (from limited testing), so I used the `py2_only` decorator to get the tests green and check if its possible to drop this test.

Reply:
Got it,  the MockS3Mixin has a lot of issues and we had a bunch of trouble with it with other tests as well.  Given that we're going to be dropping python 2 support, I think we can delete this test case if it's being flaky.

@giovannicimolin can you please delete this test in a separate PR as a follow up?

Reply:
@feanil I've created https://github.com/edx/edx-platform/pull/22618 to delete the flaky tests.